src/realTimeReporter/utils.ts

```diff
@@ -9,3 +9,53 @@ export const subscribeToEvent = (event: EVENTS, callback: (params: any) => void)
   // @ts-ignore
   process.on(event, callback);
 };
+
+export const getCodeRef = (itemName: string): string => {
+  const caller = getCaller();
+  const testPath = caller.getFileName();
+  const workingDir = process.cwd();
+
+  const codeRef = testPath
+      .replace(`${workingDir}\\`, '')
+      .replace(/\\/g, '.')
+      .replace(/\.[^/.]+$/gi, '');
+
+  return `${codeRef}.${itemName}`;
+};
+
+const getCaller = (): NodeJS.CallSite => {
+  const stack = getStack();
+
+  // Remove superfluous function calls on stack
+  stack.shift(); // getCaller --> getStack
+  stack.shift(); // your function call --> getCaller
+
+  // Return caller's caller
+  return stack[1];
+};
+
+const getStack = (): Array<NodeJS.CallSite> => {
+  // Save original Error.prepareStackTrace
+  const origPrepareStackTrace = Error.prepareStackTrace;
+
+  // Override with function that just returns `stack`
+  Error.prepareStackTrace = function (_, stack) {
```

Comment:
Why not to create own Error nested from NodeJS Error and implement this method for it without affect Node JS Error at all?

Reply:
Cannot be done it using inheritance:
1) This method is static for Error class ([reference](https://nodejs.org/api/errors.html#errors_error_capturestacktrace_targetobject_constructoropt))
2) Setting mechanism for `stack` property encapsulated in the Error class.